napari/utils/migrations.py

```diff
@@ -211,3 +211,43 @@ def __init_subclass__(cls, **kwargs):
     _OldClass.__new__.__signature__ = prealloc_signature  # type: ignore [attr-defined]
 
     return _OldClass
+
+
+class _DeprecatingDict(dict[str, Any]):
```

Comment:
This will fail if someone pass it to layer constructor using `**kwargs` 


Iteration should return only non deprecated kwargs.

Reply:
At first I did not like this suggestion because it changes the core behavior of `dict` (whereas the current approach only raises warnings under certain usage).

But I also understand that this is a critical issue. And the only time this object is created publicly is when calling `Layer.as_layer_data_tuple` or indirectly through plugins.

This approach may still causes issues for plugin authors that are iterating over the state `dict` and expecting the deprecated keys, but that's probably not many people.

Reply:
Yes. And current approach will create problem if someone do:

```
data, _type, meta = Points.as_layer_data_tuple()

new_points = Points(data, **meta)
```
As it will end with exceptions. 


And this is exactly why `test_benchamrks` fails.

Reply:
Understood. As previously mentioned and explained.